java/opendcs/src/main/java/decodes/tsdb/algo/WeightedAverage.java

```diff
@@ -0,0 +1,202 @@
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "input8")) != null && isAssigned(t) && isMissing(input8) &&
+              pr.missingAction != MissingAction.IGNORE))
+        {
+            log.debug("Skipping time slice with base time {} because of missing value for param {}",
+                    debugSdf.format(_timeSliceBaseTime), t);
+            return;
+        }
+        if (((pr = getParmRef(t = "weight1")) != null && isAssigned(t) && isMissing(weight1) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight2")) != null && isAssigned(t) && isMissing(weight2) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight3")) != null && isAssigned(t) && isMissing(weight3) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight4")) != null && isAssigned(t) && isMissing(weight4) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight5")) != null && isAssigned(t) && isMissing(weight5) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight6")) != null && isAssigned(t) && isMissing(weight6) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight7")) != null && isAssigned(t) && isMissing(weight7) &&
+              pr.missingAction != MissingAction.IGNORE)
+         || ((pr = getParmRef(t = "weight8")) != null && isAssigned(t) && isMissing(weight8) &&
+              pr.missingAction != MissingAction.IGNORE))
+        {
+            log.debug("Skipping time slice with base time {} because of missing value for param {}",
+                    debugSdf.format(_timeSliceBaseTime), t);
+            return;
+        }
+        if (!isMissing(input1) && !isMissing(weight1))
+            tot += (input1 * weight1);
+        if (!isMissing(input2) && !isMissing(weight2))
+            tot += (input2 * weight2);
+        if (!isMissing(input3) && !isMissing(weight3))
+            tot += (input3 * weight3);
+        if (!isMissing(input4) && !isMissing(weight4))
+            tot += (input4 * weight4);
+        if (!isMissing(input5) && !isMissing(weight5))
+            tot += (input5 * weight5);
+        if (!isMissing(input6) && !isMissing(weight6))
+            tot += (input6 * weight6);
+        if (!isMissing(input7) && !isMissing(weight7))
+            tot += (input7 * weight7);
+        if (!isMissing(input8) && !isMissing(weight8))
+            tot += (input8 * weight8);
+
+        // Output only if nonzero total weight
+        if (weightTotal != 0)
+        {
+            tot /= weightTotal;
+            log.trace("doAWTimeSlice baseTime={}, input1={}, weight1={}, input2={}, weight2={}, tot={}",
+                    debugSdf.format(_timeSliceBaseTime), input1, weight1, input2, weight2, tot);
```

Comment:
```suggestion
                    _timeSliceBaseTime, input1, weight1, input2, weight2, tot);
```

the long knows how to handle dates, we just left all the debugsdf objects in because ... so many and there were a few non logging usages.